rpc/src/v1/tests/helpers/snapshot_service.rs

```diff
@@ -43,6 +43,8 @@ impl TestSnapshotService {
 impl SnapshotService for TestSnapshotService {
 	fn manifest(&self) -> Option<ManifestData> { None }
 	fn supported_versions(&self) -> Option<(u64, u64)> { None }
+	fn restoration_ready(&self) -> bool { true }
+	fn completed_chunks(&self) -> Option<Vec<H256>> { Some(vec![]) }
```

Comment:
> IMO the Snapshot restoration state in the Chain Sync shouldn't exist anyway, and it should be read from the Snapshot Service when needed.

Maybe we could embed `Snapshot` in `Service` and add a bunch of methods, like `needed_chunk` to `SnapshotService` instead of `fn completed_chunks`? That way we wouldn't need to sync `Snapshot` manually.